packages/react/src/components/Forms/Select/mono-searchable.tsx

```diff
@@ -40,6 +41,17 @@ export const SelectMonoSearchable = (props: SubProps) => {
     downshiftReturn.inputValue,
   ]);
 
+  // When component is controlled, this useEffect will update the local selected item.
+  useEffect(() => {
+    if (props.downshiftProps.initialSelectedItem !== undefined) {
+      return;
+    }
+    const optionToSelect = props.options.find(
+      (option) => optionToValue(option) === props.value,
+    );
+    downshiftReturn.selectItem(optionToSelect ?? null);
+  }, [props.value]);
```

Comment:
```suggestion
  }, [downshiftReturn,
    props.downshiftProps.initialSelectedItem,
    props.options,
    props.value]);
```

Reply:
This would trigger unnecessary onChange events. We only want it to be triggered when props.value changes, at this point it will use the latest references on the other dependencies you listed. 

But maybe I don't see unwanted side-effects using only `props.value`, do you have some in mind?

Reply:
@AntoLC I agree with Nathan, I understand the point, but I find that it is not necessary to retrigger calculations for nothing in terms of performance.

Reply:
It is recommended to list in the dependencies the variables used in your `useEffect` (https://legacy.reactjs.org/docs/hooks-rules.html).
In this case, if options change (remove the option selected by example), the `useEffect` will not be retrigger, this will not be correct anymore `downshiftReturn.selectItem(optionToSelect ?? null);`

Reply:
Ok then regarding `options` I agree. But concerning `downshiftReturn` we are facing infinite useEffect calls, see:

<img width="553" alt="Capture d’écran 2023-09-15 à 14 28 29" src="https://github.com/openfun/cunningham/assets/9628870/5c9e385e-01a4-48d1-9a86-22e870fbbc6d">


```
  useEffect(() => {
    console.log("useEffect");
    if (props.downshiftProps.initialSelectedItem !== undefined) {
      return;
    }
    const optionToSelect = props.options.find(
      (option) => optionToValue(option) === props.value,
    );
    downshiftReturn.selectItem(optionToSelect ?? null);
  }, [props.value, downshiftReturn]);
```

So the rule that says that we need to add all variables used inside dependencies should not be followed as a rule of thumb imo, though I agree for the others via your demonstration. ( new fixup :) ! )